packages/edit-site/src/components/global-styles/screen-color-element.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { __experimentalColorGradientControl as ColorGradientControl } from '@wordpress/block-editor';
+/**
+ * Internal dependencies
+ */
+import ScreenHeader from './header';
+import {
+	getSupportedGlobalStylesPanels,
+	useSetting,
+	useStyle,
+	useColorsPerOrigin,
+} from './hooks';
+
+const elements = {
+	text: {
+		description: __(
+			'Set the default color used for text across the site.'
+		),
+		title: __( 'Text' ),
+	},
+	caption: {
+		description: __( 'Set the colors used for captions across the site' ),
+		title: __( 'Captions' ),
+	},
+	button: {
+		description: __(
+			'Set the default colors used for buttons across the site.'
+		),
+		title: __( 'Buttons' ),
+	},
+};
+
+function ScreenColorElement( { name, element } ) {
+	const supports = getSupportedGlobalStylesPanels( name );
+	const colorsPerOrigin = useColorsPerOrigin( name );
+	const [ solids ] = useSetting( 'color.palette', name );
+	const [ areCustomSolidsEnabled ] = useSetting( 'color.custom', name );
+	let [ isBackgroundEnabled ] = useSetting( 'color.background', name );
+	const [ isTextEnabled ] = useSetting( 'color.text', name );
+
+	let textColorElementSelector = 'elements.' + element + '.color.text';
+	const backgroundColorElementSelector =
+		'elements.' + element + '.color.background';
+
+	let hasElementColor = false;
+
+	switch ( element ) {
+		case 'button':
+			hasElementColor =
+				supports.includes( 'buttonColor' ) &&
+				isBackgroundEnabled &&
+				( solids.length > 0 || areCustomSolidsEnabled );
+			break;
+		case 'text':
+			hasElementColor =
+				supports.includes( 'color' ) &&
+				isTextEnabled &&
+				( solids.length > 0 || areCustomSolidsEnabled );
+			textColorElementSelector = 'color.text';
```

Comment:
Is this the only place we mutate `textColorElementSelector`? If so, would it work to do the calculation when we define that variable, and switch to a `const`?

Alternately, what if we moved the selector to be a key of the element in the `elements` object above (so, `elements.text.textColorElementSelector`?